agent/mibgroup/agentx/protocol.c

```diff
@@ -1643,6 +1643,8 @@ agentx_parse(netsnmp_session * session, netsnmp_pdu *pdu, u_char * data,
          * expects to find the context in the PDU's context field.  Therefore we
          * need to copy the context into the PDU's context fields.  */
         if (pdu->community_len > 0 && pdu->contextName == NULL) {
+            if (strlen(pdu->community) != pdu->community_len)
```

Comment:
Please add a comment that explains that using strlen() is safe because snmp_clone_mem() '\0'-terminates its output or use memchr() instead of strlen().